### PR TITLE
🐛  Source Shopify: add `discount applications` to orders stream

### LINKED ISSUE #19348: Source Shopify: Line Item Discount Application
## Environment
- **Airbyte version**:  0.40.18
- **OS Version / Instance**: Debian 11
- **Deployment**: Docker in GCP
- **Source Connector and version**: Shopify 0.2.0
- **Destination Connector and version**: BigQuery Denomralized 1.2.5
- **Step where error happened**: Sync job

## Current Behavior
In the orders stream, each struct in the line_items array has an attribute called discount_allocations, which is always empty.

## Expected Behavior
The attribute should be called discount_applications and contain an array of discounts applied to the line item, as per the [Shopify API documentation](https://shopify.dev/api/admin-rest/2022-10/resources/order).

## Logs
Not relevant

## Steps to Reproduce
Connect a Shopify instance with BigQuery denormalized and run a sync.

## Are you willing to submit a PR?
I can't develop at this point.

Thanks!

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: https://github.com/airbytehq/airbyte/issues/19348

## How
- added `discount_applications` property to the `orders` schema
- updated `SAT` config with new format + high strictness lvl

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
For certain cases the `Refresh Source Schema` is required to pull latest schema updates for `orders` stream.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [X] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [X] Code reviews completed
- [X] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by darynaishchenko — APPROVED** [review]

**Review by girarda — APPROVED** [review]

File: airbyte-integrations/connectors/source-shopify/acceptance-test-config.yml [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
-      # 'abandoned_checkouts' stream holds data up to 1 month.
-      empty_streams: ["abandoned_checkouts", "balance_transactions"]
+    tests:
+      - config_path: "secrets/config.json"
+        timeout_seconds: 7200
+        expect_records:
+          path: "integration_tests/expected_records.txt"
+        empty_streams:
+          - name: "shop"
+            bypass_reason: "The stream contains sensitive personal information. Skipped."
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-shopify/acceptance-test-config.yml:34]: Could you please let me know in DM on Slack what's exactly sensitive? Can we change these information on our  Shopify account to make these less sensitive?

**Review by alafanechere — APPROVED** [review]
Thanks for your efforts.
I'm approving the `acceptance-test-config.yml` change, but would love not to bypass the `shop` stream, can you check if we could change the shop information for less sensitive data?

File: airbyte-integrations/connectors/source-shopify/acceptance-test-config.yml (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/acceptance-test-config.yml:34]: It's the info about "Who" created the `Shop` and his personal info like: Phone, Email, etc. 
That info should be real I  guess to confirm with the owner if something is happening.

**Review by lazebnyi — APPROVED** [review]

**Comment by bazarnov** [comment]
Accidentally Auto-merged. The publish is following up here: https://github.com/airbytehq/airbyte/pull/21485

**Pull request merged**